packages/core/src/discovery.js

```diff
@@ -88,21 +88,42 @@ function waitForDiscoveryNetworkIdle(page, options) {
 
 // Creates an initial resource map for a snapshot containing serialized DOM
 function parseDomResources({ url, domSnapshot }) {
-  if (!domSnapshot) return new Map();
-  let isHTML = typeof domSnapshot === 'string';
-  let { html, resources = [] } = isHTML ? { html: domSnapshot } : domSnapshot;
-  let rootResource = createRootResource(url, html);
+  const map = new Map();
+  if (!domSnapshot) return map;
+  let allResources = new Set();
+
+  if (Array.isArray(domSnapshot)) {
```

Comment:
Any specific reason to bifurcate this ? wouldnt converting a single snapshot entry to an array work here ? Its basically doing same thing per snapshot